src/Uri.php

```diff
@@ -437,9 +437,9 @@ public function withScheme($scheme)
 
     public function withUserInfo($user, $password = null)
     {
-        $info = $user;
+        $info = rawurlencode($user);
```

Comment:
this breaks bc with existing code as it doubles escapes stuff that people passed encoded already. so we need to make sure to not double escape. this logic is already happening for the path, query etc.
Preventing double escaping makes it ambigous when using usernames/passwords that resemble percent-encoded values like `pass%23word`. So for the generic case you still need to escape manually before calling the function. But at least this does not break bc and is consistent with path encoding handling.

Reply:
see also https://github.com/zendframework/zend-diactoros/blob/0c108ccdf5a1db0d976c13cb9140775dc828c205/src/Uri.php#L561-L570

Reply:
OK